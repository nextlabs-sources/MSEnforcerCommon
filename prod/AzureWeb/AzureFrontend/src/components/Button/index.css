.btn-loading::before {
    content: '';
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    background-color: rgba(255, 255, 255, .4);    
}
.btn-spinner {
    display: inline-block;
    vertical-align: top;
    position: relative;
    color: #fff;
    width: 14px;
    height: 14px;
    margin-left: -14px;
}
.btn-spinner::before {
    content: '';
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    box-sizing: border-box;
    border-width: 2px;
    border-style: solid;
    border-color: #fff transparent transparent transparent;
    border-radius: 14px;
    animation: spinning ease-in-out .8s infinite;    
}