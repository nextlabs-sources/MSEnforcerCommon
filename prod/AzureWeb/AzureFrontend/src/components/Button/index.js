import React from 'react'
import propTypes from 'prop-types'
import './index.css'

const Button = ({ className, onClick, loading = false, children }) => {

    const btnClass = loading ? `btn-loading ${className}` : className

    return (
        <button className={btnClass} style={{ position: 'relative' }} onClick={!loading ? onClick : null}>
            {
                loading ? <i className='btn-spinner'></i> : null
            }
            {
                loading ? <span style={{marginLeft: '14px', verticalAlign: 'top'}}>{children}</span> : <span>{children}</span>
            }
        </button>
    )
}

export default Button

Button.propTypes = {
    className: propTypes.string.isRequired,
    onClick: propTypes.func.isRequired,
    loading: propTypes.bool,
    children: propTypes.node.isRequired,
}