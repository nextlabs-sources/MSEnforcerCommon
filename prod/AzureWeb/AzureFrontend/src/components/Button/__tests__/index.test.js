import React from 'react'
import renderer from 'react-test-renderer'
import Button from '../index'


describe('Button', () => {
    it('should render loading element <i/>', () => {
        const btnClickHandler = jest.fn()
        const tree = renderer.create(<Button className={'test-class'} loading={true} onClick={btnClickHandler}>TestButton</Button>).toJSON()

        expect(tree).toMatchSnapshot()
        expect(tree.props.onClick).toBe(null)
    })
    
    it('should not render loading element <i/>', () => {
        const btnClickHandler = jest.fn()
        const tree = renderer.create(<Button className={'test-class'} loading={false} onClick={btnClickHandler}>TestButton</Button>).toJSON()

        tree.props.onClick()

        expect(tree).toMatchSnapshot()
        expect(btnClickHandler.mock.calls.length).toBe(1) 
    })
})
