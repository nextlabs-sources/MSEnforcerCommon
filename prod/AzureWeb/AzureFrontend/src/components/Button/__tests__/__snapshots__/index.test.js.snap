// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should not render loading element <i/> 1`] = `
<button
  className="test-class"
  onClick={[Function]}
  style={
    Object {
      "position": "relative",
    }
  }
>
  <span>
    TestButton
  </span>
</button>
`;

exports[`Button should render loading element <i/> 1`] = `
<button
  className="btn-loading test-class"
  onClick={null}
  style={
    Object {
      "position": "relative",
    }
  }
>
  <i
    className="btn-spinner"
  />
  <span
    style={
      Object {
        "marginLeft": "14px",
        "verticalAlign": "top",
      }
    }
  >
    TestButton
  </span>
</button>
`;
